test/e2e/Benchmark.t.sol

```diff
@@ -0,0 +1,195 @@
+        CowProtocol.emptySettleWithInteractions(onlyApprove);
+    }
+
+    /// @param extraDataSize The size of the padding data that will be included
+    /// in an interaction to pad the settlement. This can be approximately be
+    /// the calldata size of a `settle` interaction without flash loans.
+    /// @param name An identifier that will be used to give a name to the
+    /// snapshot for this run.
+    function flashLoanSettleWithExtraData(uint256 extraDataSize, string memory name) private {
+        uint256 loanedAmount = 1; // Just one wei of the token
+        // We assume that the fees aren't larger than the traded amount.
+        uint256 fees = loanedAmount;
+
+        // We'll pay the fees from the buffers.
+        uint256 settlementBalance = token.balanceOf(address(Constants.SETTLEMENT_CONTRACT));
+        assertGt(settlementBalance, fees);
+
+        // Start preparing the settlement interactions.
+        ICowSettlement.Interaction[] memory interactionsWithFlashLoan = new ICowSettlement.Interaction[](4);
+        // Transfer tokens out from the settlement wrapper to the settlement
+        // contract. In practice, the funds will be sent to a different address,
+        // but for the purpose of the test this should have an equivalent cost
+        // of modifying a fresh storage slot.
+        interactionsWithFlashLoan[0] = CowProtocolInteraction.transferFrom(
+            token, address(solverWrapper), address(Constants.SETTLEMENT_CONTRACT), loanedAmount
+        );
+        // Approve repayment. Under some assumptions, this could be done once
+        // per call.
+        interactionsWithFlashLoan[1] =
+            CowProtocolInteraction.wrapperApprove(solverWrapper, token, lender, loanedAmount + fees);
+        // Prepare flash-loan repayment.
+        interactionsWithFlashLoan[2] =
+            CowProtocolInteraction.transfer(token, address(solverWrapper), loanedAmount + fees);
+        // Padding transaction. It does nothing but increase the cost of
+        // executing flash-loan in a way that is compatible with the execution
+        // of other interactions with on-chain liquidity to settle an order.
+        // In practice, the gas cost of these interactions will be different,
+        // but that's not something that should change between the case with the
+        // flash loan or without it.
+        // As a first approximation, using the actual calldata size of a normal
+        // call to `settle` seems reasonable, though it slightly overestimates
+        // the actual cost (some of the calldata used in a settlement is due to
+        // the ABI encoding of the transaction in a format compatible with a
+        // `settle()` call, which has to be done regardless). However, we
+        // consider the impact of this extra data overall small.
+        interactionsWithFlashLoan[3] =
+            ICowSettlement.Interaction({target: address(0), value: 0, callData: new bytes(extraDataSize)});
```

Comment:
Does it make a difference that this is 0 bytes calldata (in practice our calldata will be non-zero bytes)